### PR TITLE
Batch registration spreadsheet is only reset if the set experiment has changed

### PR DESCRIPTION
**Pull request by Steffengreiner**

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java [hunk 1]
```diff
@@ -103,14 +104,19 @@ public boolean isInputValid() {
...
+    this.experiment = experiment;
     experimentName.setText(experiment.getName());
+    SampleRegistrationSpreadsheet.setExperimentMetadata(experiment);
   }
 
   public List<SampleRegistrationContent> getContent() {
     return sampleInformationLayoutHandler.getContent();
   }
 
+  public Experiment getExperiment() {
```
Comment (sven1103) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java:116]: Since experiment is an entity not an DTO, this is not side effect free. Why is this needed?

**Review by sven1103 — CHANGES_REQUESTED** [review]

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java (hunk 1 above)
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java]: Good point I'll expose only the experimentID

**Review by sven1103 — APPROVED** [review]
